registry.go

```diff
@@ -57,18 +57,38 @@ func Login(ctx context.Context, store Store, reg *remote.Registry, cred auth.Cre
 
 // Logout provides the logout functionality given the registry name.
 func Logout(ctx context.Context, store Store, registryName string) error {
-	registryName = mapHostname(registryName)
+	registryName = mapStoreRegistryName(registryName)
 	if err := store.Delete(ctx, registryName); err != nil {
 		return fmt.Errorf("unable to delete the credential for %s: %w", registryName, err)
 	}
 	return nil
 }
 
-func mapHostname(hostname string) string {
-	// The Docker CLI expects that the 'docker.io' credential
-	// will be added under the key "https://index.docker.io/v1/"
-	if hostname == "docker.io" {
+// Credential returns a Credential() function that can be used by auth.Client.
+func Credential(store Store) func(context.Context, string) (auth.Credential, error) {
+	return func(ctx context.Context, reg string) (auth.Credential, error) {
+		reg = mapAuthenticationRegistryName(reg)
+		if reg == "" {
+			return auth.EmptyCredential, nil
+		}
+		return store.Get(ctx, reg)
+	}
+}
+
+// The Docker CLI expects that the 'docker.io' credential
+// will be added under the key "https://index.docker.io/v1/"
+func mapStoreRegistryName(registry string) string {
+	if registry == "docker.io" {
+		return "https://index.docker.io/v1/"
+	}
+	return registry
+}
+
+// It is expected that the traffic targetting "registry-1.docker.io"
+// will be redirected to "https://index.docker.io/v1/"
+func mapAuthenticationRegistryName(target string) string {
```

Comment:
Should rename `target` to `hostname` for better readability.

Reply:
changed.